decidim-core/app/helpers/decidim/external_domain_helper.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Decidim
+  module ExternalDomainHelper
+    def highlight_domain(url)
+      parts = url.match %r{^(([a-z]+):)?//([^/]+)(/.*)?$}
```

Comment:
This fails right now in case the URL is in unexpected format.

We should add the following here:
```ruby
raise InvalidUrlError unless parts
```

Also, we need to define the `Decidim::InvalidUrlError` which we can do below this module as follows:
```ruby
  end # <-- end for ExternalDomainHelper

  class InvalidUrlError < StandardError; end
end # <-- end for the namespace of the helper
```

After that, rescue from that error in the controller and handle the unexpected case (see the other comment).